### PR TITLE
Miles per hour option added

### PR DESCRIPTION
**Pull request by ShaleenM**
I have added MPH capability to the application in response to Issue "Miles per hour please #16".
1. Current Speed changes to miles per hour.
2. Distance if greater than 1 miles changes to miles.
3. Average speed changes to miles per hour.
4. Max Speed changes to miles per hour.

Added checkbox option in settings to switch from KMPH to MPH and back.

### CODE REVIEW

**Comment by flyingrub** [comment]
Wow totaly forgot this one, i will try to review it soon.

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 1]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
     public void addDistance(double distance){
         distanceM = distanceM + distance;
         distanceKm = distanceM / 1000f;
+        distanceMiles = distanceM / 1609.34;
     }
 
-    public SpannableString getDistance(){
+    public SpannableString getDistance(boolean isMPH){
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: this variable could be passed in Data constructor and stored there.

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 2]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
...
         SpannableString s;
-        if (distanceKm < 1) {
-            s = new SpannableString(String.format("%.0f", distanceM) + "m");
-            s.setSpan(new RelativeSizeSpan(0.5f), s.length() - 1, s.length(), 0);
-        }else{
+        if(isMPH && distanceMiles >= 1){
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: I suggest you to do a first if with only the isMPH and then a nested one with the condition related to it. Your current implementation can lead to errors.

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 3]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
...
+            s = new SpannableString(String.format("%.0f", distanceM) + "m");
+            s.setSpan(new RelativeSizeSpan(0.4f), s.length() - 1, s.length(), 0);
         }
         return s;
     }
 
-    public SpannableString getMaxSpeed() {
-        SpannableString s = new SpannableString(String.format("%.0f", maxSpeed) + "km/h");
-        s.setSpan(new RelativeSizeSpan(0.5f), s.length() - 4, s.length(), 0);
+    public SpannableString getMaxSpeed(boolean isMPH) {
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: Same as before

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 4]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
...
+        else
+            s = new SpannableString(String.format("%.0f", maxSpeed) + "km/h");
+
+        s.setSpan(new RelativeSizeSpan(0.4f), s.length() - 4, s.length(), 0);
         return s;
     }
 
-    public SpannableString getAverageSpeed(){
-        double average = ((distanceM / (time / 1000)) * 3.6);
+    public SpannableString getAverageSpeed(boolean isMPH){
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: here too

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 5]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
...
+                s = new SpannableString(0 + "km/h");
+            }
+            s.setSpan(new RelativeSizeSpan(0.4f), s.length() - 4, s.length(), 0);
         }
-        s.setSpan(new RelativeSizeSpan(0.5f), s.length() - 4, s.length(), 0);
         return s;
     }
 
-    public SpannableString getAverageSpeedMotion(){
+    public SpannableString getAverageSpeedMotion(Boolean isMPH){
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: again

File: app/src/main/java/fly/speedmeter/grub/GpsServices.java [hunk 6]
```diff
@@ -84,8 +84,9 @@ public void updateNotification(boolean asData){
                 .setSmallIcon(R.drawable.ic_notification)
                 .setContentIntent(contentIntent);
 
+
         if(asData){
-            builder.setContentText(String.format(getString(R.string.notification), data.getMaxSpeed(), data.getDistance()));
+            builder.setContentText(String.format(getString(R.string.notification), data.getMaxSpeed(true), data.getDistance(true)));
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/GpsServices.java]: why don't you get the value from the pref ?

File: app/src/main/java/fly/speedmeter/grub/MainActivity.java [hunk 7]
```diff
@@ -191,7 +193,9 @@ protected void onResume() {
         else
             tempspeed = currentSpeed.getText();
 
-        if (sharedPreferences.getBoolean("isMPH", false))
```
Comment (flyingrub) [inline app/src/main/java/fly/speedmeter/grub/MainActivity.java]: why don't you write `isMPH = sharedPreferences.getBoolean("isMPH", false);` only ?

File: app/src/main/java/fly/speedmeter/grub/Data.java (hunk 1 above)
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: Using an enum for `Metric` and `Imperial` may be more clear.

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 8]
```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
...
+    public SpannableString getAverageSpeed(boolean isMPH){
+
         SpannableString s;
-        if (time > 0){
-            s = new SpannableString(String.format("%.0f", average) + "km/h");
 
-        }else{
-            s = new SpannableString(0 + "km/h");
+        if(isMPH){
+            double average = ((distanceM /time ) * 2.23694);
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: Odd formatting.

File: app/src/main/java/fly/speedmeter/grub/MainActivity.java [hunk 9]
```diff
@@ -182,11 +187,29 @@ protected void onResume() {
...
+            tempspeed = "0kmph";
+        else
+            tempspeed = currentSpeed.getText();
+
+        updateIsMPH();
+
+        if (isMPH)
+            tempspeed = tempspeed.subSequence(0, tempspeed.length() - 4) + "mph ";
+        else
+            tempspeed = tempspeed.subSequence(0, tempspeed.length() - 4) + "kmph";
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/MainActivity.java]: Do these various strings need localization?  Repeated throughout.

File: app/src/main/java/fly/speedmeter/grub/Data.java [hunk 10]
```diff
@@ -134,5 +166,12 @@ public long getTime() {
     public void setTime(long time) {
         this.time = time;
     }
-}
 
+    public void setMaxSpeed(double maxSpeed){
+        this.maxSpeed = maxSpeed;
+    }
+
+    public double getMaxSpeedDouble(){
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/Data.java]: Why not `getMaxSpeed`?

File: app/src/main/java/fly/speedmeter/grub/MainActivity.java [hunk 11]
```diff
@@ -50,15 +50,18 @@
...
     protected void onCreate(Bundle savedInstanceState) {
+        Log.d("Metric" , "In OnCreate");
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
         data = new Data(onGpsServiceUpdate);
 
-        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+//        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+        sharedPreferences = getSharedPreferences("speedmeter", MODE_PRIVATE);
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/MainActivity.java]: Why make this change and leave the commented code?

File: app/src/main/java/fly/speedmeter/grub/MainActivity.java [hunk 12]
```diff
@@ -248,7 +273,13 @@ public void onLocationChanged(Location location) {
...
         if (location.hasSpeed()) {
             progressBarCircularIndeterminate.setVisibility(View.GONE);
-            SpannableString s = new SpannableString(String.format("%.0f", location.getSpeed() * 3.6) + "km/h");
+            Log.d("Metric", "Contains isMPH ?? " + sharedPreferences.contains("isMPH"));
+            Log.d("Metric" , " isMPH == " + sharedPreferences.getBoolean("isMPH", false));
+            SpannableString s;
+            if(isMPH)
+                s = new SpannableString(String.format("%.0f", location.getSpeed() * 2.23694) + "mph ");
+            else
+                s = new SpannableString(String.format("%.0f", location.getSpeed() * 3.6) + "km/h");
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/MainActivity.java]: Different formattings of kmph and km/h.

File: app/src/main/java/fly/speedmeter/grub/MainActivity.java [hunk 13]
```diff
@@ -326,6 +357,25 @@ public void onBackPressed(){
...
+    public void updateIsMPH(){
+        if(isMPH == sharedPreferences.getBoolean("isMPH", false))
+            return;
+
+        isMPH = sharedPreferences.getBoolean("isMPH", false);
+
+        if(isMPH)
+            data.setMaxSpeed(data.getMaxSpeedDouble() * 0.621371);
+        else
+            data.setMaxSpeed(data.getMaxSpeedDouble() * 1.60934);
```
Comment (andrewgaul) [inline app/src/main/java/fly/speedmeter/grub/MainActivity.java]: I don't understand what is happening here -- SpeedMeter should always store speeds in kilometers per second and only display in miles per hour.

**Comment by flyingrub** [comment]
https://github.com/flyingrub/SpeedMeter/commit/62173d52af5a3de46de0620964fb97422ff90c31 fixes it.

**Pull request closed without merging**